src/pages/Transaction/TransactionComp/TransactionComp.tsx

```diff
@@ -18,18 +24,89 @@ const handleCellbaseInputs = (inputs: Cell[], outputs: Cell[]) => {
 }
 
 export const TransactionComp = ({ transaction }: { transaction: Transaction }) => {
-  const { transactionHash, displayInputs, displayOutputs, blockNumber, isCellbase } = transaction
+  const txHash = transaction.transactionHash
+  const location = useLocation()
+  const history = useHistory()
+  const queryParams = new URLSearchParams(location.search)
+  const inputCellsPageNumber = queryParams.get('inputCellsPageNumber')
+  const inputCellsPageSize = queryParams.get('inputCellsPageSize')
+  const outputCellsPageNumber = queryParams.get('outputCellsPageNumber')
+  const outputCellsPageSize = queryParams.get('outputCellsPageSize')
 
-  const inputs = handleCellbaseInputs(displayInputs, displayOutputs)
+  const DEFAULT_PAGE_SIZE = 10
+  const inputCellsPageLimit = inputCellsPageSize ? Number(inputCellsPageSize) : DEFAULT_PAGE_SIZE
+  const outputCellsPageLimit = outputCellsPageSize ? Number(outputCellsPageSize) : DEFAULT_PAGE_SIZE
+  const [inputCellsPage, setInputCellsPage] = useState(inputCellsPageNumber ? Number(inputCellsPageNumber) : 1)
+  const handleInputCellsPageChange = (page: number) => {
+    setInputCellsPage(page)
+    queryParams.set('inputCellsPageNumber', page.toString())
+    history.replace(`${location.pathname}?${queryParams.toString()}`)
+  }
+  const txInputsQuery = useQuery(['transactionInputs', txHash, inputCellsPage], async () => {
```

Comment:
`inputCellsPageLimit` is also a condition of the query, which should be included in the query key, otherwise the query result will be cached mistakenly